yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -422,3 +429,42 @@ func (p *ParserIssueDetector) genericIssues(query string) ([]QueryIssue, error)
 
 	return result, nil
 }
+
+func (p *ParserIssueDetector) getJsonbColumns() []string {
+	var jsonColumns []string
+	for _, mp := range p.columnsWithUnsupportedIndexDatatypes {
```

Comment:
let's not re-use `columnsWithUnsupportedIndexDatatypes`  to get jsonb columns here.  If in the future, jsonb index gets supported, we'll have to remove it from that list, but we'll need it for detecting jsonb-related issues. 

Can we use a different variable here? 

Also, this means that we need to have a single instance of `parserIssueDetector` for both analyzeSchema as well as assess-migration AND that the unsupportedqueryconstructs are fetched AFTER the schema issues have been detected, correct? Can you verify that this is indeed happening?

Reply:
> Also, this means that we need to have a single instance of parserIssueDetector for both analyzeSchema as well as assess-migration AND that the unsupportedqueryconstructs are fetched AFTER the schema issues have been detected, correct? Can you verify that this is indeed happening?

yes correct, we would need that because we rely on exported schema details for other things, e.g. datatypes (UDT, array etc..)  so it is still the case that `parserIssueDetector` needs to be the same for analyze and assessment and run any other detection in assessment after exported schema is Parsed (ParseRequiredDDL()) .

Reply:
But if required all this can be separated if we run the `ParseRequiredDDL` separately on exported schema for assessment.

Reply:
I think it's okay for now, it's a fair assumption to make, I suppose. just add a comment in code at the point where we're calling parserIssueDetector to get unsupported query constructs.